commons/src/main/scala/dagr/commons/reflect/ReflectionUtil.scala

```diff
@@ -290,7 +292,11 @@ object ReflectionUtil {
     */
   private[reflect] def typedValueFromString(resultType: Class[_], unitType: Class[_], value: String*): Try[Any] = Try {
     if (ReflectionUtil.isCollectionClass(resultType)) {
-      val typedValues = value.map(v => buildUnitFromString(unitType, v).get).asInstanceOf[Seq[java.lang.Object]]
+      val typedValues = value.indexOf(SpecialEmptyOrNoneToken) match {
```

Comment:
So .. given that for both `Option` and `Collection` types providing one or more values replaces the default values, I think it would be sufficient (and clearer/simpler/safer) here to check and see if the input `value` has a single value in it which matches the `SpecialEmptyOrNoneToken` or not.  If it does, return `Seq.empty()`, else convert the entire array.